nav2_costmap_2d/include/nav2_costmap_2d/inflation_layer.hpp

```diff
@@ -133,11 +127,13 @@ class InflationLayer : public Layer
    * @param src_y The y coordinate of the source cell
    * @return
    */
-  inline double distanceLookup(int mx, int my, int src_x, int src_y)
+  inline double distanceLookup(
+    unsigned int mx, unsigned int my, unsigned int src_x,
+    unsigned int src_y)
   {
-    unsigned int dx = abs(mx - src_x);
-    unsigned int dy = abs(my - src_y);
-    return cached_distances_[dx][dy];
+    unsigned int dx = (mx > src_x) ? mx - src_x : src_x - mx;
```

Comment:
This is nearly identical to the implementation of abs (https://code.woboq.org/userspace/glibc/stdlib/abs.c.html), please revert for readability.

Reply:
I did this as initially `distanceLookup(int mx, int my, int src_x, int src_y)` was this but where ever it was called its arguments were `unsigned int`:
https://github.com/ros-planning/navigation2/blob/d21cf3366b06e3a63014e411708c2cbae2ae3647/nav2_costmap_2d/plugins/inflation_layer.cpp#L235

So it had to make 2 casts one from unsigned int to int then back to unsigned int when assigning to `dx` and `dy`.
This way there are no casts.

Reply:
Ah ok, I can live with that